services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -154,15 +161,18 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     val urlEncoder = URLEncoder.encode(path.toString, "UTF-8")
     val commitsUrl: String =
       s"${config.url}projects/${project.repository.get.value}/repository/files/${urlEncoder}"
-    httpClient.get[List[FileCommit]](url = commitsUrl, headers = config.token).map {
-      case Response(_, SuccessResponseBody(commitsSeq), _) => Right(commitsSeq)
-      case Response(_, FailureResponseBody(error), _) =>
-        Left(
-          VersioningException.FileException(
-            s"Could not take the file commits. ResponseBody: ${error}"
+    httpClient
+      .get[List[FileCommit]](url = commitsUrl, headers = config.token)
+      .map {
+        case Response(_, SuccessResponseBody(commitsSeq), _) => Right(commitsSeq)
+        case Response(_, FailureResponseBody(error), _) =>
+          Left(
+            VersioningException.FileException(
+              s"Could not take the file commits. ResponseBody: ${error}"
+            )
           )
-        )
-    }
+      }
+      .recover { case e: Throwable => Left(VersioningException.HttpException("recover " + e.getMessage)) }
```

Comment:
It is okay that you decided to recover from such errors, but please make a more meaningful message. "recover XXX" sounds strange.

Reply:
This is just a line, from @manvicha request that was forgotten, I have just restored it

Reply:
First of all, this is your request and you are responsible for every single line :)
Secondly, can you give me a link to this PR, please?

Reply:
https://github.com/EpamLifeSciencesTeam/cmwl_pipeline/pull/83/files that is

Reply:
This PR has nothing to do with recovering from errors. It only introduces new types of errors.